src/vi/nettruyen/src/eu/kanade/tachiyomi/extension/vi/nettruyen/NetTruyen.kt

```diff
@@ -18,9 +25,163 @@ class NetTruyen : WPComics(
      * That makes both sites always return un-relevant results when searching should return empty.
      */
     override fun searchMangaParse(response: Response): MangasPage {
-        if (response.request.url.queryParameter(name = queryParam) == null) {
-            return MangasPage(mangas = listOf(), hasNextPage = false)
+        if (response.request.url.toString().endsWith("/$searchPath")) {
+            return MangasPage(mangas = emptyList(), hasNextPage = false)
         }
         return super.searchMangaParse(response)
     }
+
+    // Advanced search
+    override fun searchMangaRequest(page: Int, query: String, filters: FilterList): Request {
+        val advancedGenreFilter = filters.filterIsInstance<AdvancedGenreFilter>().firstOrNull()
+        val advancedStatusFilter = filters.filterIsInstance<AdvancedStatusFilter>().firstOrNull()
+        val chaptersNumFilter = filters.filterIsInstance<ChaptersNumFilter>().firstOrNull()
+        val genderFilter = filters.filterIsInstance<GenderFilter>().firstOrNull()
+        val orderFilter = filters.filterIsInstance<OrderFilter>().firstOrNull()
+
+        if (query.isBlank() &&
+            (
+                advancedGenreFilter?.state?.any { it.isIncluded() || it.isExcluded() } == true ||
+                    advancedStatusFilter != null && advancedStatusFilter.state > 0 ||
+                    chaptersNumFilter != null && chaptersNumFilter.state > 0 ||
+                    genderFilter != null && genderFilter.state > 0 ||
+                    orderFilter != null && orderFilter.state > 0
+                )
```

Comment:
why such a convoluted logic? just put default values if the filter isn't applied

Reply:
There are currently 2 working search methods:
- Old method: query string with status & genre selection
- New advanced method: genre tristate, gender, chapter number.. but won't support query.

This code is intended to invoke the new advanced search method if any of the new filters are applied, fall-back to old search if no new filter is used. On the other hand, if the user doesn't enter a query string but still uses old search's filters, we can't decide the search method solely depends on the query string alone.

Reply:
does the new advanced method on the site doesn't support filters that are available in old method?

Reply:
Only the genre & status filter is common between 2 methods, but the value used for genre filters are different. Also some genres is available in the new search but not in the old one. I don't think it's possible to merge 2 methods:
- Historical:
  - Old: Name: Historical - Value: historical
  - New: Name: Lich su - Value: 17
- Yuri: only in new search
- Yaoi: only in new search

Reply:
imo this would be confusing for the users, which filter works in which condition...

Reply:
I agree it does look confused. How about add a string to clarify like this:
<img width="324" alt="image" src="https://github.com/keiyoushi/extensions-source/assets/16017808/39cd1e6f-28b3-4db9-b5b2-f5ef80c50a7a">

And simplify condition to only use old search if query string is entered.